### PR TITLE
docs: update about ports and ssl cert for virtualhost feature

### PR DESCRIPTION
**Pull request by thotz**
The users need to careful about port and ssl certs while consuming virtual host feature for rgw.

Fixes: #14087

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 1]
```diff
@@ -159,6 +159,9 @@ The hosting settings allow you to host buckets in the object store on a custom D
 
 * `dnsNames`: a list of DNS names to host buckets on. These names need to valid according RFC-1123. Otherwise it will fail. Each endpoint requires wildcard support like [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards). Do not include the wildcard itself in the list of hostnames (e.g., use "mystore.example.com" instead of "*.mystore.example.com"). Add all the hostnames like openshift routes otherwise access will be denied, but if the hostname does not support wild card then virtual host style won't work those hostname. By default cephobjectstore service endpoint and custom endpoints from cephobjectzone is included. The feature is supported only for Ceph v18 and later versions.
 
+!!! Note
+    The user need to confirm the port exposed by wildcard supported endpoint should be same that of rgw kubernetes service. Most provides like ingress loadbalancer or openshift routes does the same. If the SSL enabled for `CephObjectStore` then the `dnsNames` should include the endpoints which exposes the secure port and user need to update the SSL certs used by the RGW contains that domain.
```
Comment (travisn) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:163]: Could you rephrase the paragraph? I don't understand it.

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md (hunk 1 above)
Comment (thotz) [reply Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:163]: I want to cover following:

The port-defined domains in `dnsNames` accessing rgw should be same as `cephobjectstore.Spec`. By default ingress loadbalancer or openshift routes uses same port as the service which they are exposing 

If secureport is defined in `cephobjectstoreSpec`, then dnsNames contain domains which uses the secure port. And this domain need approved by the SSL certs which RGW is using

Comment (travisn) [reply Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:163]: What is a "port-defined domain"? I thought the port was independent from the dns name.
I understand better now what you are saying. Can you update the doc paragraph now instead of just the comment?

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 2]
```diff
@@ -159,6 +159,14 @@ The hosting settings allow you to host buckets in the object store on a custom D
 
 * `dnsNames`: a list of DNS names to host buckets on. These names need to valid according RFC-1123. Otherwise it will fail. Each endpoint requires wildcard support like [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards). Do not include the wildcard itself in the list of hostnames (e.g., use "mystore.example.com" instead of "*.mystore.example.com"). Add all the hostnames like openshift routes otherwise access will be denied, but if the hostname does not support wild card then virtual host style won't work those hostname. By default cephobjectstore service endpoint and custom endpoints from cephobjectzone is included. The feature is supported only for Ceph v18 and later versions.
 
+!!! Important considerations:
```
Comment (travisn) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:162]:
```suggestion
!!! important
```

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 3]
```diff
@@ -159,6 +159,14 @@ The hosting settings allow you to host buckets in the object store on a custom D
...
+
+* Domains mentioned here should listen to same port as RGW kubernetes service.
```
Comment (travisn) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:164]:
```suggestion
* Domains added to `dnsNames` should listen to the same port as the RGW Kubernetes service.
```

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 4]
```diff
@@ -159,6 +159,14 @@ The hosting settings allow you to host buckets in the object store on a custom D
...
+
+* If RGW kubernetes service listen to both `secure` and `non-secure` port, include domains which uses secure port.
```
Comment (travisn) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:166]: To be included in the "important" block, these lines must be indented four spaces, and I'm not sure it supports bullets for proper rendering. @BlaineEXE do you know? Perhaps we just make a paragraph instead of bullets.
```suggestion
    If the RGW Kubernetes service listens to both `secure` and `non-secure` ports, include domains which use a secure port.
```

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md (hunk 4 above)
Comment (BlaineEXE) [reply Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:166]: It won't support bullets inside the callout. This should just be part of the doc.

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 5]
```diff
@@ -158,6 +158,10 @@ The [zone](../../Storage-Configuration/Object-Storage-RGW/ceph-object-multisite.
 The hosting settings allow you to host buckets in the object store on a custom DNS name, enabling virtual-hosted-style access to buckets similar to AWS S3 (https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html).
 
 * `dnsNames`: a list of DNS names to host buckets on. These names need to valid according RFC-1123. Otherwise it will fail. Each endpoint requires wildcard support like [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards). Do not include the wildcard itself in the list of hostnames (e.g., use "mystore.example.com" instead of "*.mystore.example.com"). Add all the hostnames like openshift routes otherwise access will be denied, but if the hostname does not support wild card then virtual host style won't work those hostname. By default cephobjectstore service endpoint and custom endpoints from cephobjectzone is included. The feature is supported only for Ceph v18 and later versions.
+Please consider below points while setting this option:
+  * Domains added to `dnsNames` should listen to the same port as the RGW Kubernetes service.
```
Comment (BlaineEXE) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:162]: I'm starting to get confused. What is "the RGW Kubernetes service" in this sentence?

If you are referring to the service that Rook creates automatically for RGWs, then we should instead refer to this from the user-facing view, which is the CephObjectStore spec and instead say that domains should listen on the same port configured in the CephObjectStore gateway spec.

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 6]
```diff
@@ -158,6 +158,10 @@ The [zone](../../Storage-Configuration/Object-Storage-RGW/ceph-object-multisite.
...
+  * If the RGW Kubernetes service listens to both `secure` and `non-secure` ports, include domains which use a secure port.
+  * If TLS is enabled with RGW, either include domains in `dnsNames` to the certificate or set `insecureSkipVerify` to true in the kubernetes secret containing TLS cert.
```
Comment (BlaineEXE) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:164]: I'm confused about the wording here. Could you attempt to use more clear wording?

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 7]
```diff
@@ -158,6 +158,10 @@ The [zone](../../Storage-Configuration/Object-Storage-RGW/ceph-object-multisite.
...
 
 * `dnsNames`: a list of DNS names to host buckets on. These names need to valid according RFC-1123. Otherwise it will fail. Each endpoint requires wildcard support like [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards). Do not include the wildcard itself in the list of hostnames (e.g., use "mystore.example.com" instead of "*.mystore.example.com"). Add all the hostnames like openshift routes otherwise access will be denied, but if the hostname does not support wild card then virtual host style won't work those hostname. By default cephobjectstore service endpoint and custom endpoints from cephobjectzone is included. The feature is supported only for Ceph v18 and later versions.
+Please consider below points while setting this option:
+  * Domains added to `dnsNames` should listen to the same port as the RGW Kubernetes service.
+  * If the RGW Kubernetes service listens to both `secure` and `non-secure` ports, include domains which use a secure port.
```
Comment (BlaineEXE) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:163]: Again, I am confused about "RGW Kubernetes service."

When users read text that is in a code block `like this` they will naturally interpret those to be field names related to configuration. That isn't the case here. Please reword to specifically reference `port` and `securePort` for user benefit.

Also, I think the phrase "include domains which use a secure port" might not be clear enough instruction for users. I think it would be more clear to state "domains input into `dnsNames` must only specify secure port." This will help users know that they shouldn't put insecure domain addresses in the list, which is what we are trying to clarify. We could even plainly say, "do not put domains which only listen on the insecure `port` into the `dnsNames` list."

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
Please also fix the markdown issues flagged by our doc linter.

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 8]
```diff
@@ -158,6 +158,10 @@ The [zone](../../Storage-Configuration/Object-Storage-RGW/ceph-object-multisite.
...
+Please consider below points while setting this option:
+    * Domains added to `dnsNames` should listen to the same port mentioned in `CephObjectStoreSpec`.
+    * If the user specified both `port` and `securePort` in the `CephObjectStoreSpec`, then domains input into `dnsNames` must only specify `securePort`.
+    * If TLS is enabled with RGW, either TLS certificate include to verify the domains in `dnsNames` or user need to set `insecureSkipVerify` to true in the kubernetes secret.
```
Comment (BlaineEXE) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:164]: This doesn't read clearly in multiple places, and I'm having trouble figuring out what users need to do. 

Is this what is intended?

"If TLS is enabled, the TLS certificate must include the domains in `dnsNames`, or `insecureSkipVerify` must be set on the CephObjectStore."

Comment (BlaineEXE) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:164]: I can't tell if this section is intended to be part of the paragraph that begins with the bullet `* dnsNames` above, or if it is intended to be a separate paragraph.

As it is, the text "Please consider..." immediately follows the `dnsNames` bullet, which implies it should be the same content, but it also is indented on the left, which implies it should be a separate paragraph.

The 3 bullets are indented, which implies there is intent to have this be part of the `dnsNames` bullet.

IMO, this section is short enough that this will be easier to read for users if it is a separate paragraph. To format it as such, please add an empty line above "Please consider..." and un-indent the 3 bullets.

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
